source/elements/oneDAL/source/algorithms/clustering/kmeans.rst

```diff
@@ -18,109 +20,197 @@ minimize the objective function
 where :math:`d(x_i, C)` is the distance from :math:`x_i` to the closest center
 in :math:`C`.
 
+
+-------------
+Usage example
+-------------
+::
+
+   onedal::kmeans::model run_training(const onedal::table& data,
+                                      const onedal::table& initial_centroids) {
+
+      const auto kmeans_desc = onedal::kmeans::desc<float>{}
+         .set_cluster_count(10)
+         .set_max_iteration_count(50)
+         .set_accuracy_threshold(1e-4);
+
+      const auto result = onedal::train(kmeans_desc, data, initial_centroids);
+
+      print_table("labels", result.get_labels());
+      print_table("centroids", result.get_model().get_centroids());
+      print_value("objective", result.get_objective_function_value());
+
+      return result.get_model();
+   }
+
+   onedal::table run_inference(const onedal::kmeans::model& model,
+                               const onedal::table& new_data) {
+
+      const auto kmeans_desc = onedal::kmeans::desc<float>{}
+         .set_cluster_count(model.get_cluster_count());
+
+      const auto result = onedal::infer(kmeans_desc, model, new_data);
+
+      print_table("labels", result.get_labels());
+   }
+
+
 ---
 API
 ---
 
 Descriptor
 ----------
-.. code-block:: c++
-
-   namespace onedal::kmeans {
+::
 
    namespace method {
-   struct lloyd {};
-   using by_default = lloyd;
+      struct lloyd {};
+      using by_default = lloyd;
    } // namespace method
 
    template <typename Float = float,
              typename Method = method::by_default>
    class desc {
    public:
-      double get_gamma() const;
+      desc();
+
+      int64_t get_cluster_count() const;
+      int64_t get_max_iteration_count() const;
       double get_accuracy_threshold() const;
-      std::int64_t get_cluster_count() const;
-      std::int64_t get_max_iteration_count() const;
 
-      desc& set_gamma(double);
+      desc& set_cluster_count(int64_t);
+      desc& set_max_iteration_count(int64_t);
       desc& set_accuracy_threshold(double);
-      desc& set_cluster_count(std::int64_t);
-      desc& set_max_iteration_count(std::int64_t);
    };
 
-   } // namespace onedal::kmeans
+cluster_count ``int64_t`` ``default = 2``
+   The number of clusters.
+
+   Invariants
+      | ``cluster_count > 0``
+
+max_iterations ``int64_t`` ``default = 100``
+   The maximum number of iterations.
+
+   Invariants
+      | ``max_iterations > 0``
+
+accuracy_threshold ``double`` ``default = 0.0``
```

Comment:
it would be easier for implementer consume those parameters (I also keep in mind other, more complex algorithms), if they are introduced/described somewhere in the beginning
also, we need to specify, what this threshold means, that is what stop criteria, what would happen, if threshold is set to zero

Reply:
Added description of stop criteria